packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx

```diff
@@ -340,105 +308,20 @@ const ChallengeTeamWizard: React.FunctionComponent<
-      setRegisterChallengeSuccess(false)
-      await registerChallengeTeam(
-        {
-          challengeId: challenge.id,
-          teamId: String(teamId),
-        },
-        accessToken,
-      )
-        .then(() => {
-          setRegisterChallengeSuccess(true)
-        })
-        .catch((err: ErrorResponse) => {
-          setErrorMessage(`${msg}: ${err.reason}`)
-          setConfirming(false)
-        })
-    } else {
-      setErrorMessage(`${msg}: Invalid team.`)
-      setConfirming(false)
-    }
-  }
-
-  // CREATE_NEW_TEAM: Invite a comma-delimited list of emails to join the team
-  const handleInviteTeamMembers = async (
-    teamId: string | number,
-    invitees: string,
-  ) => {
-    if (!invitees.length) {
-      setInviteMembersSuccess(true)
-      return
-    }
-    const msg = 'Error inviting members'
-    if (teamId) {
-      setErrorMessage(undefined)
-      setInviteMembersSuccess(false)
-      const emails = invitees.split(',')
-      const errors: string[] = []
-      for (const inviteeEmail of emails) {
-        await createMembershipInvitation(
-          { teamId: String(teamId), inviteeEmail: inviteeEmail.trim() },
-          accessToken,
-        ).catch(() => {
-          errors.push(inviteeEmail.trim())
-        })
-      }
-
-      if (errors.length) {
-        setErrorMessage(`${msg}: ${errors.join(', ')}`)
-        setConfirming(false)
-      } else {
-        setInviteMembersSuccess(true)
-      }
-    } else {
-      setErrorMessage(`${msg}: Invalid team.`)
-      setConfirming(false)
-    }
-  }
-
-  // CREATE_NEW_TEAM: Create a new team to join the challenge
-  const handleCreateTeam = async () => {
-    if (newTeam && newTeam.name && newTeam.name.length > 1) {
-      setStep({ ...step, confirmEnabled: false })
-      setConfirming(true)
-      setErrorMessage(undefined)
-      await createTeam(
-        { name: newTeam.name, description: newTeam.description },
-        accessToken,
-      )
-        .then(response => {
-          setCreatedNewTeam(true)
-          setSelectedTeam(response)
-          // Add newly created team to challenge
-          handleRegisterChallengeTeam(response.id)
-          // Invite emails to new team
-          handleInviteTeamMembers(response.id, newTeam.invitees)
-        })
-        .catch((err: ErrorResponse) => {
-          setConfirming(false)
-          setErrorMessage(`Error creating team: ${err.reason}`)
-          setStep({ ...step, confirmEnabled: true })
-        })
-    }
-  }
-
   const hide = () => {
     setErrorMessage(undefined)
     setCreatedNewTeam(false)
-    setSelectedTeam(undefined)
+    setSelectedTeamId(undefined)
     setStep({ ...step, nextEnabled: false })
     onClose()
   }
 
   const onConfirmHandlerMap: Record<string, () => Promise<void> | void> = {
-    CREATE_NEW_TEAM: handleCreateTeam,
+    CREATE_NEW_TEAM: () => {
+      if (createChallengeTeamRef.current) {
+        createChallengeTeamRef.current.submit()
+      }
+    },
```

Comment:
When `Confirm` is clicked, use the ref to invoke `submit` in the component, which will trigger creating + registering the team.